src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -395,20 +423,31 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 			}
 		}
 
-		samplesAppender, err := appender.SamplesAppender(opts)
+		result, err := appender.SamplesAppender(opts)
 		if err != nil {
-			return err
+			multiErr = multiErr.Add(err)
+		}
+
+		if result.IsDropPolicyApplied {
+			iter.SetCurrentMetadata(ts.Metadata{true})
 		}
 
 		for _, dp := range datapoints {
-			err := samplesAppender.AppendGaugeTimedSample(dp.Timestamp, dp.Value)
+			err := result.SamplesAppender.AppendGaugeTimedSample(dp.Timestamp, dp.Value)
 			if err != nil {
-				return err
+				// If we see an error break out so we can try processing the
+				// next datapoint.
+				multiErr = multiErr.Add(err)
+				break
 			}
 		}
 	}
 
-	return iter.Error()
+	if err := iter.Error(); err != nil {
+		multiErr = multiErr.Add(err)
+	}
```

Comment:
nit: multiErr checks for nil internally, can just do `multiErr = multiErr.Add(iter.Error())`